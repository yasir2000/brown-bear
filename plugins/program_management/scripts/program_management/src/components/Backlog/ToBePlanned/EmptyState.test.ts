/**
 * Copyright (c) Enalean, 2021 - Present. All Rights Reserved.
 *
 * This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import type { ShallowMountOptions } from "@vue/test-utils";
import { shallowMount } from "@vue/test-utils";
import EmptyState from "./EmptyState.vue";
import { createProgramManagementLocalVue } from "../../../helpers/local-vue-for-test";

describe("EmptyState", () => {
    let component_options: ShallowMountOptions<EmptyState>;

    it("Displays the empty state for Backlog unplanned section", async () => {
        component_options = {
            propsData: {
                project_public_name: "Public name",
                project_short_name: "short-name",
            },
            localVue: await createProgramManagementLocalVue(),
        };

        const wrapper = shallowMount(EmptyState, component_options);
        expect(wrapper.element).toMatchSnapshot();
    });
});
