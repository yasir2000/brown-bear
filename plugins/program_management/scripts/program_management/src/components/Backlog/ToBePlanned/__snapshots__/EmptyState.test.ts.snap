// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState Displays the empty state for Backlog unplanned section 1`] = `
<section
  class="empty-state-page"
  project_public_name="Public name"
  project_short_name="short-name"
>
  <p
    class="empty-state-text"
    data-current-language="en_US"
    data-msgid="There is nothing to plan"
  >
    There is nothing to plan
  </p>
</section>
`;
